src/library_pthread.js

```diff
@@ -404,6 +413,10 @@ var LibraryPThread = {
     {{{ makeSetValue('__num_logical_cores', 0, 'cores', 'i32') }}};
   },
 
+  emscripten_transfer_canvas: function() {
+    Module['canvasTransferRequested'] = true;
```

Comment:
Not sure about this property, really. I just could not think of a better way to make this work. Alternative was to create a emscripten-specific version of `pthread_create` taking additional options. That seemed too obtuse, but maybe you have a better idea?